### PR TITLE
[risk=low][no ticket] AccessModuleNameMapper

### PR DESCRIPTION
**Pull request by jmthibault79**
Add an `AccessModuleNameMapper` to convert between API entity `AccessModule`, `DbAccessModuleName`, `BypassTimeTargetProperty`, and `BadgeName`

TODO in #6325
* update `AccessModule` enum values to match `DbAccessModuleName`
* rename `AccessModule` to `AccessModuleName` or similar

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/BypassTimeTargetProperty.kt [hunk 1]
```diff
@@ -2,10 +2,10 @@ package org.pmiops.workbench.actionaudit.targetproperties
 
 enum class BypassTimeTargetProperty
 constructor(override val propertyName: String) : SimpleTargetProperty {
-    DATA_USE_AGREEMENT_BYPASS_TIME("data_use_agreement_bypass_time"),
-    COMPLIANCE_TRAINING_BYPASS_TIME("compliance_training_time"),
-    CT_COMPLIANCE_TRAINING_BYPASS_TIME("ct_compliance_training_time"),
-    ERA_COMMONS_BYPASS_TIME("era_commons_bypass_time"),
-    TWO_FACTOR_AUTH_BYPASS_TIME("two_factor_auth_bypass_time"),
-    RAS_LINK_LOGIN_GOV("ras_link_login_gov_bypass_time")
+    DATA_USER_CODE_OF_CONDUCT("data_use_agreement_bypass_time"),
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/BypassTimeTargetProperty.kt:5]: changing the **strings** would mean changing the recorded audit values.  That needs more thought.

File: api/src/main/java/org/pmiops/workbench/access/AccessUtils.java [hunk 2]
```diff
@@ -1,70 +1,15 @@
...
+import org.pmiops.workbench.db.model.DbAccessModule.DbAccessModuleName;
 import org.pmiops.workbench.db.model.DbAccessTier;
 import org.pmiops.workbench.exceptions.NotFoundException;
-import org.pmiops.workbench.model.AccessModule;
 
 /** Utilities for RW Access related functionalities. */
 public class AccessUtils {
   private AccessUtils() {}
 
-  private static final BiMap<AccessModule, AccessModuleName> CLIENT_TO_STORAGE_ACCESS_MODULE =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessUtils.java:17]: replaced by AccessModuleNameMapper

File: api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/BypassTimeTargetProperty.kt (hunk 1 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/BypassTimeTargetProperty.kt:5]: We're free to rename the enum values though - so now they match AccessModuleName

**Review by evrii — APPROVED** [review]
Looks good to me. All the tests in api/src/test all passed for me. The Dev App Server also started without issue.

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java [hunk 3]
```diff
@@ -154,7 +156,8 @@ public boolean isModuleCompliant(DbUser dbUser, DbAccessModuleName accessModuleN
     DbAccessModule dbAccessModule =
         getDbAccessModuleOrThrow(dbAccessModulesProvider.get(), accessModuleName);
     // if the module is not required, the user is always compliant
-    if (!isModuleRequiredInEnvironment(storageAccessModuleToClient(dbAccessModule.getName()))) {
+    if (!isModuleRequiredInEnvironment(
+        accessModuleNameMapper.storageAccessModuleToClient(dbAccessModule.getName()))) {
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java]: This simply converts an enum value of type A to type B.  "storage to client" is a common naming pattern we use for converting between DB entities and API entities.

That said, I found a way to save us a database call so I'm updating this line now.

File: api/src/test/java/org/pmiops/workbench/access/UserAccessModuleMapperTest.java [hunk 4]
```diff
@@ -16,7 +16,12 @@
 import org.springframework.context.annotation.Import;
 import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
 
-@Import({FakeClockConfiguration.class, UserAccessModuleMapperImpl.class, CommonMappers.class})
+@Import({
+  AccessModuleNameMapperImpl.class,
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/access/UserAccessModuleMapperTest.java:20]: This is a common Spring gotcha: class dependencies have changed for objects in this test file, so the test now requires having this impl bean imported to pass.

This comes up a lot in API unit testing, so it's good to learn what these errors look like.  Try removing this import and I expect you'll see a big spammy error.  There should be a reference to this service (not the impl!) but you have to hunt for it.

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java [hunk 5]
```diff
@@ -151,12 +153,14 @@ public boolean hasUserSignedACurrentDucc(DbUser targetUser) {
 
   @Override
   public boolean isModuleCompliant(DbUser dbUser, DbAccessModuleName accessModuleName) {
-    DbAccessModule dbAccessModule =
-        getDbAccessModuleOrThrow(dbAccessModulesProvider.get(), accessModuleName);
     // if the module is not required, the user is always compliant
-    if (!isModuleRequiredInEnvironment(storageAccessModuleToClient(dbAccessModule.getName()))) {
+    if (!isModuleRequiredInEnvironment(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:157]: slight logic change: now checks for presence in environment before checking the DB

**Pull request merged**